app/models/squix/print_job_wrapper.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Squix
+  # Squix::PrintJobWrapper
+  class PrintJobWrapper
+    include ActiveModel::Model
+
+    attr_accessor :printer_name, :label_template_name, :label_template_id, :labels, :copies, :errors
+
+    # validate
+
+    def initialize(printer_name, label_template_name, label_template_id, labels, copies)
+      @printer_name = printer_name
+      @label_template_name = label_template_name
+      @label_template_id = label_template_id
+      @labels = labels
+      @copies = copies.to_i
+      @errors = []
+    end
+
+    def print
+      # return false unless valid?
+
+      case printer_type
+      when 'Toshiba'
+        print_to_toshiba
+      when 'Squix'
+        print_to_squix
+      else
+        errors << "Printer type #{printer_type} for printer #{printer_name} not recognised."
+        false
+      end
+    end
+
+    def printer_type
```

Comment:
probably better to just return the printer as we will need other attributes.

Reply:
I'll make this change in https://github.com/sanger/print_my_barcode/pull/69

Reply:
have done what I can until GPL-831-2 is merged